app/src/main/java/com/example/advanceduiapp/data/PageData.kt

```diff
@@ -0,0 +1,3 @@
+package com.example.advanceduiapp.data
+
+data class PageData(val title:String, val resourceId: Int, override val type: ViewType = ViewType.Page): ItemData
```

Comment:
do you need to have a `type` here if you use different types in different classes? So it means that you can figure out that types are different just by checking the real type(class) of the object

Reply:
Well, I had a similar approach before. If we wanna receive a single data list in MultipleViewRecyclerAdapter then it`s easier when every item knows its viewType code. It has to be an integer value returned by getItemViewType(position: Int). So the only alternative that comes to my mind is to implement a function in the view model that will eventually do the same just trying to find out what is the viewType of a given Item